logstreams/src/main/java/io/zeebe/logstreams/processor/ReProcessingStateMachine.java

```diff
@@ -118,53 +113,107 @@ private ReProcessingStateMachine(
     this.streamProcessorName = context.getName();
     this.eventFilter = context.getEventFilter();
     this.logStreamReader = context.getLogStreamReader();
+    this.isReadOnlyProcessor = context.isReadOnlyProcessor();
+    this.producerId = context.getId();
 
     this.streamProcessor = streamProcessor;
     this.zeebeDb = zeebeDb;
-    this.updateStateRetryStrategy = new RecoverableRetryStrategy(actor);
+    this.updateStateRetryStrategy = new EndlessRetryStrategy(actor);
+    this.processRetryStrategy = new EndlessRetryStrategy(actor);
     this.abortCondition = abortCondition;
   }
 
   // current iteration
   private long lastSourceEventPosition;
   private ActorFuture<Void> recoveryFuture;
-
   private LoggedEvent currentEvent;
   private EventProcessor eventProcessor;
   private ZeebeDbTransaction zeebeDbTransaction;
 
-  ActorFuture<Void> startRecover(long lastSourceEventPosition) {
-    this.lastSourceEventPosition = lastSourceEventPosition;
+  ActorFuture<Void> startRecover(final long snapshotPosition) {
     recoveryFuture = new CompletableActorFuture<>();
-    reprocessNextEvent();
+
+    final long startPosition = logStreamReader.getPosition();
+
+    LOG.info("Start scanning the log for error events.");
+    lastSourceEventPosition = scanLog(snapshotPosition);
+    LOG.info(
+        "Processor {} starts reprocessing, until last source event position {}",
```

Comment:
This log statement is only true if `lastSourceEventPosition` is greater than `snapshotPosition`. It could be confusing to write it in other cases.

Reply:
Ah moved this stmt, this was the log statement we saw in the morning right :sweat_smile: